storage/remote/write_handler.go

```diff
@@ -395,7 +399,21 @@ func (h *writeHandler) appendV2(app storage.Appender, req *writev2.Request, rs *
 		var ref storage.SeriesRef
 
 		// Samples.
-		for _, s := range ts.Samples {
+		for i, s := range ts.Samples {
+			// CT only needs to be ingested for the first sample, it will be considered
+			// out of order for the rest.
+			if i == 0 && ts.CreatedTimestamp != 0 && h.ingestCTZeroSample {
```

Comment:
Do we need to do this in loop if you have to do it on the start? Can we do this is before looping here? (same for histograms).

Reply:
Would it make sense to add the same for histograms once we get #14694 merged? Just to be sure we don't implement different behaviors